ssl/ssl_rsa.c

```diff
@@ -1035,3 +1035,124 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
     BIO_free(bin);
     return ret;
 }
+
+static int ssl_set_cert_and_key(SSL *ssl, SSL_CTX *ctx, X509 *x509, EVP_PKEY *privatekey,
+                                STACK_OF(X509) *chain, int override)
+{
+    int ret = -1;
+    size_t i;
+    int j;
+    int rv;
+    CERT *c = ssl ? ssl->cert : ctx->cert;
+    STACK_OF(X509) *dup_chain = NULL;
+    EVP_PKEY *pubkey = NULL;
+
+    /* Do all security checks before anything else */
```

Comment:
I might consider an ossl_assert() that exactly one of ssl and ctx are NULL.

Reply:
I believe this idiom is used in other places, and no ossl_assert() is present. But this is also an internal API, so I'm a little reluctant to add it.